### ISSUE TITLE
Pipe name conflict

### ISSUE THREAD

**Issue by afnpires** [angular/angular#13569]
**I'm submitting a ...**  (check one with "x")
```
[] bug report => search github for a similar issue or PR before submitting
[x] feature request
[ ] support request => Please do not submit support request here, instead see https://github.com/angular/angular/blob/master/CONTRIBUTING.md#question
```

**Current behavior**

When having 2 pipes with the same name, exported in two different modules, there's no way to specify which one of them is used nor a error is throw specifying that the current pipe already exists.

```javascript
@Pipe({ name: 'highlight' })
export class HighlightPipe implements PipeTransform {
//...
}
```

As an example I had a HiglightPipe in my solution and [ng2-bootstrap](https://github.com/valor-software/ng2-bootstrap) was providing other HiglightPipe. 
The ng2-bootstrap was the last module imported and was the used pipe.

**Expected behavior**

An error specifying that the pipe with that name is already in use or a way to specify which modules pipe to use.

* **Angular version:** 2.2.1

**Comment by pkozlowski-opensource** [comment]
I agree that we should throw / report an error in this case. Having said this libraries should _really_ prefix their stuff....

**Comment by afnpires** [comment]
@pkozlowski-opensource that's what I ended up doing to my own pipe.

**Comment by LayZeeDK** [comment]
> I agree that we should throw / report an error in this case. Having said this libraries should _really_ prefix their stuff....

Maybe Angular should have lead by example by prefixing the common pipes.

**Comment by sharshuv-quotient** [comment]
Thumbs up for that. My team member just wasted a couple of hours trying to understand why his pipe didn't work and we ended up going through all imported modules and their corresponding pipes to find a collision.

**Comment by LayZeeDK** [comment]
@sharshuv-quotient

I suggest using SCAMs and migrating to Standalone Components, Directives, and Pipes in the future.

**Comment by decherneyge** [comment]
I am curious why it is not recommended that pipes within your codebase should be prefixed the same way components should be prefixed. Here is a link to the pipe naming docs: https://angular.io/guide/styleguide#pipe-names and here is a link to the Component custom prefix docs: https://angular.io/guide/styleguide#component-custom-prefix

Here is the relevant snippet from the Component custom prefix docs that I think also applies to pipes

> Do use a custom prefix for a component selector. For example, the prefix toh represents Tour of Heroes and the prefix admin represents an admin feature area.
>
> Do use a prefix that identifies the feature area or the application itself.
>
> Why?
>Prevents element name collisions with components in other applications and with native HTML elements.
>
>Why?
>Makes it easier to promote and share the component in other applications.
>
>Why?
>Components are easy to identify in the DOM.

I feel like all of these points also apply to pipes. For example, it seems better to have an application prefix on a pipe so that when I see one in a template, I know "oh that is a pipe from this app and not a native angular pipe." 

`<span>{{someLongTextVariable | ellipsis}}</span>`

is that a built in angular thing or does that come from this codebase ? 

`<span>{{someLongTextVariable | toh-ellipsis}}</span>`

Oh the toh app has a ellipsis pipe that it uses to control its rules for truncating long text. 


I know there is nothing that prevents a codebase from defining its own preferred practices, but I feel like there is good reason for this to be a community standard practice as well. I am curious if others agree/disagree.
